nemo/collections/nlp/models/language_modeling/megatron_gpt_peft_models.py

```diff
@@ -55,16 +60,34 @@ def init_peft_modules(self):
         Randomly initialize the peft params and add them to the appropriate modules.
         """
         assert len(self.peft_name_keys) > 0, "peft_name_keys have not been set no PEFT modules will be added"
+        assert not self.mcore_gpt or hasattr(
+            self, 'name_key_to_mcore_mixins'
+        ), "some of the PEFT modules are not supported in megatron core mode yet."
         assert len(self.name_key_to_cfg) > 0, "name_key_to_cfg has not been set no PEFT modules will be added"
         logging.info(f"Before adding PEFT params:\n{self.summarize()}")
-        for _, module in self.named_modules():
-            if isinstance(module, adapter_mixins.AdapterModuleMixin):
+        for name, module in self.named_modules():
+            # if there's an infinite loop here, check that the added adapter modules doesn't use the same key as mcore targets.
+            if self.mcore_gpt:
                 for peft_key in self.peft_name_keys:
-                    peft_cfg = self.name_key_to_cfg[peft_key]
-                    if model_utils.import_class_by_path(peft_cfg._target_) in module.get_accepted_adapter_types():
-                        module.add_adapter(
-                            name=peft_key, cfg=peft_cfg,
-                        )
+                    for mcore_target, mcore_mixin in self.name_key_to_mcore_mixins[peft_key]:
+                        if name.split(".")[-1] == mcore_target:
```

Comment:
should I do a suffix match or a full regex match?

e.g. the template for lora mcore target will be 
```
"model\.decoder\.layers\.[0-9]+\.self_attention\.linear_qkv"
```